demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -192,6 +213,35 @@ static uint16_t globalSubscribePacketIdentifier = 0U;
  */
 static uint16_t globalUnsubscribePacketIdentifier = 0U;
 
+/**
+ * @brief MQTT packet type received from the MQTT broker.
+ *
+ * @note Only on receiving incoming SUBACK or UNSUBACK, this
+ * variable is updated. For MQTT packets PUBACK and PINGRESP, the variable is
+ * not updated since there is no need to specifically wait for it in this demo.
+ * A single variable suffices as this demo uses single task and requests one operation
+ * (of SUBSCRIBE or UNSUBSCRIBE) at a time before expecting response from
+ * the broker. Hence it is not possible to receive multiple packets of type
+ * SUBACK and UNSUBACK in a single call of #prvWaitForPacket.
+ * For a multi task application, consider a different method to wait for the packet, if needed.
+ */
+static uint16_t globalPacketTypeReceived = 0U;
+
+/**
+ * @brief A pair containing a topic filter and its SUBACK status.
+ */
+typedef struct topicFilterContext
+{
+    const char * pcTopicFilter;
+    MQTTSubAckStatus_t xSubAckStatus;
+} topicFilterContext_t;
+
+/**
+ * @brief An array containing the context of a SUBACK; the SUBACK status
+ * of a filter is updated when the event callback processes a SUBACK.
+ */
+static topicFilterContext_t xTopicFilterContext;
```

Comment:
An array?

Reply:
`SubscribeToTopic` only expects a single topic filter so an array is not needed in this case

Reply:
I meant to say that the documentation (that mentions array) does not meet the definition..